mobilecoind-json/src/main.rs

```diff
@@ -0,0 +1,440 @@
+    // requested value. This can be used as-is in the transfer call below, or the value can be
+    // modified.
+    Ok(Json(JsonReadRequestResponse {
+        receiver: JsonPublicAddress {
+            view_public_key: hex::encode(receiver.get_view_public_key().get_data().to_vec()),
+            spend_public_key: hex::encode(receiver.get_spend_public_key().get_data().to_vec()),
+            fog_fqdn: String::from(receiver.get_fog_fqdn()),
+        },
+        value: resp.get_value().to_string(),
+        memo: resp.get_memo().to_string(),
+    }))
+}
+
+#[derive(Deserialize, Serialize)]
+struct JsonTransferResponse {
+    key_images: Vec<String>,
+    tombstone: u64,
+}
+
+/// Performs a transfer from a monitor and subaddress. The public keys and amount are in the POST data.
+#[post(
+    "/monitors/<monitor_hex>/<subaddress_index>/transfer",
+    format = "json",
+    data = "<transfer>"
+)]
+fn transfer(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    subaddress_index: u64,
+    transfer: Json<JsonReadRequestResponse>,
+) -> Result<Json<JsonTransferResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    // Decode the keys
+    let mut view_public_key = RistrettoPublic::new();
+    view_public_key.set_data(
+        hex::decode(&transfer.receiver.view_public_key)
+            .map_err(|err| format!("Failed to decode view key hex: {}", err))?,
+    );
+    let mut spend_public_key = RistrettoPublic::new();
+    spend_public_key.set_data(
+        hex::decode(&transfer.receiver.spend_public_key)
+            .map_err(|err| format!("Failed to decode spend key hex: {}", err))?,
+    );
+
+    // Reconstruct the public address as a protobuf
+    let mut public_address = PublicAddress::new();
+    public_address.set_view_public_key(view_public_key);
+    public_address.set_spend_public_key(spend_public_key);
+    public_address.set_fog_fqdn(transfer.receiver.fog_fqdn.clone());
+
+    // Generate an outlay
+    let mut outlay = mc_mobilecoind_api::Outlay::new();
+    outlay.set_receiver(public_address);
+    outlay.set_value(
+        transfer
+            .value
+            .parse::<u64>()
+            .map_err(|err| format!("Failed to parse amount: {}", err))?,
+    );
+
+    // Send the payment request
+    let mut req = mc_mobilecoind_api::SendPaymentRequest::new();
+    req.set_sender_monitor_id(monitor_id);
+    req.set_sender_subaddress(subaddress_index);
+    req.set_outlay_list(RepeatedField::from_vec(vec![outlay]));
+
+    let resp = state
+        .mobilecoind_api_client
+        .send_payment(&req)
+        .map_err(|err| format!("Failed to send payment: {}", err))?;
+
+    // The receipt from the payment request can be used by the status check below
+    let receipt = resp.get_sender_tx_receipt();
+    Ok(Json(JsonTransferResponse {
```

Comment:
I think for future-proofing `JsonTransferResponse` should contain a field `sender_tx_receipt` similarly to how the GRPC api does it. Not doing so would mean whoever uses this would run into issues if we ever decide to include additional information such as the receiver receipts.